libs/service-portal/health/src/screens/Medicine/TabPanes/MedicineCalculator.tsx

```diff
@@ -0,0 +1,394 @@
+import {
+  Text,
+  Box,
+  Pagination,
+  Table as T,
+  FilterInput,
+  Button,
+  LoadingDots,
+} from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import { messages } from '../../../lib/messages'
+import { CONTENT_GAP_LG, SECTION_GAP } from '../constants'
+import { IntroHeader, m } from '@island.is/service-portal/core'
+import { useEffect, useState } from 'react'
+import { useDebounce, useWindowSize } from 'react-use'
+import {
+  useGetDrugCalculationMutation,
+  useGetDrugsQuery,
+} from '../Medicine.generated'
+import {
+  RightsPortalCalculatorRequestInput,
+  RightsPortalDrug,
+  RightsPortalDrugCalculatorResponse,
+} from '@island.is/api/schema'
+import * as styles from './Medicine.css'
+import { EmptyTable } from '../components/EmptyTable/EmptyTable'
+import { DrugRow } from '../components/DrugRow/DrugRow'
+import { useIntl } from 'react-intl'
+
+const DEFAULT_PAGE_NUMBER = 1
+const DEFAULT_PAGE_SIZE = 8
+
+export const MedicineCalulator = () => {
+  const { formatMessage } = useLocale()
+  const [search, setSearch] = useState('')
+  const [debouncedSearch, setDebouncedSearch] = useState('')
+  const [pageNumber, setPageNumber] = useState(DEFAULT_PAGE_NUMBER)
+  const [hasCalculated, setHasCalculated] = useState(false)
+  const [hoveredDrug, setHoveredDrug] = useState(-1)
+  const [selectedDrugList, setSelectedDrugList] = useState<
+    RightsPortalCalculatorRequestInput[]
+  >([])
+
+  const intl = useIntl()
+
+  useDebounce(
+    () => {
+      setDebouncedSearch(search)
+      setPageNumber(DEFAULT_PAGE_NUMBER)
+    },
+    500,
+    [search],
+  )
+
+  const {
+    data: drugs,
+    loading: drugsLoading,
+    error,
+  } = useGetDrugsQuery({
+    variables: {
+      input: {
+        pageNumber: pageNumber - 1,
+        nameStartsWith: debouncedSearch,
+        limit: DEFAULT_PAGE_SIZE,
+      },
+    },
+  })
+
+  const [drugCalcQuery] = useGetDrugCalculationMutation()
+
+  const [calculatorResults, setCalculatorResults] =
+    useState<RightsPortalDrugCalculatorResponse | null>(null)
+
+  const SHOW_TABLE = !!(
+    debouncedSearch.length > 0 && drugs?.rightsPortalDrugs.data.length
+  )
+
+  const CALCULATOR_DISABLED = selectedDrugList.length === 0
+
+  const handleCalculate = () => {
+    if (!hasCalculated) setHasCalculated(true)
+    const input = {
+      drugCalculatorRequestDTO: {
+        drugs: selectedDrugList.map((d) => ({
+          lineNumber: d.lineNumber,
+          nordicCode: d.nordicCode,
+          price: d.price,
+          units: d.units,
+        })),
+      },
+    }
+    if (input.drugCalculatorRequestDTO.drugs.length) {
+      drugCalcQuery({
+        variables: {
+          input: input,
+        },
+        onCompleted: (data) => {
+          setCalculatorResults(data.rightsPortalDrugsCalculator)
+        },
+      })
+    }
+  }
+
+  useEffect(() => {
+    if (hasCalculated) {
+      handleCalculate()
+    }
+  }, [JSON.stringify(selectedDrugList)])
```

Comment:
Why stringify every render? Feels unnecessesarily expensive when you can do maybe on stringification on selectedDrugList change instead of every single render

Reply:
Ah yes, I was stringifying so that I could compare the array values, switched to spreading.